### DISCUSSION TITLE
Is it possible to capture `stdout` from `to_process.run_sync` ?

### DISCUSSION THREAD

**Discussion by steve-marmalade** [agronholm/anyio#650] [Q&A] [answered]
The synchronous function that I am running with `run_sync` prints some valuable information to stdout during execution. Currently I am not able to see this -- is there any way to capture it and return it to the caller?

**Comment by steve-marmalade** [comment]
I just saw this in the docs:

> sys.stdin and sys.stdout, sys.stderr are redirected to /dev/null so [print()](https://docs.python.org/3/library/functions.html#print) and [input()](https://docs.python.org/3/library/functions.html#input) won’t work

Are there any workarounds that people are using?

**Comment by agronholm** [comment, marked as answer]
Right now, there is no official way to do this. Future releases may add stdout redirection so at least stdout and stderr will work as expected.

**Reply by steve-marmalade** [reply]
Thanks very much for the quick response!
